File: code/community/1306409493092896773/improving-pronunciation-indian-names-deepgram-tts.py
import os
import requests
import json

# Set the API key from environment variables
api_key = os.getenv('DEEPGRAM_API_KEY')

# Set the API endpoint
api_endpoint = 'https://api.deepgram.com/v1/speak'

# Construct text with phonetic spelling to improve pronunciation
text = "Prasanna (p-r-ah-s-a-n-n-a), Mangesh (m-uh-ng-ey-sh), Anil (ah-n-ee-l) are example names."

# JSON payload
payload = {
    'text': text,
    'voice': 'en-US'  # Assuming the use of a US voice model
}

# Send POST request to Deepgram TTS API
response = requests.post(
    api_endpoint,
    headers={
        'Authorization': f'Bearer {api_key}',
        'Content-Type': 'application/json'
    },
    data=json.dumps(payload)
)

# Check if the response is successful
if response.status_code == 200:
    print("Audio generated successfully.")
else:
    print(f"Failed to generate audio: {response.status_code}")

File: code/community/1306409493092896773/improving-pronunciation-indian-names-deepgram-tts.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

class DeepgramTTS
{
    private static readonly HttpClient client = new HttpClient();

    public static async Task Main(string[] args)
    {
        // Set the API key from environment variables
        string apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");

        // Set the API endpoint
        string apiEndpoint = "https://api.deepgram.com/v1/speak";

        // Construct text with phonetic spelling to improve pronunciation
        string text = "Prasanna (p-r-ah-s-a-n-n-a), Mangesh (m-uh-ng-ey-sh), Anil (ah-n-ee-l) are example names.";

        // JSON payload
        var payload = new
        {
            text = text,
            voice = "en-US" // Assuming the use of a US voice model
        };

        string jsonPayload = JsonConvert.SerializeObject(payload);

        client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", apiKey);

        // Send POST request to Deepgram TTS API
        var content = new StringContent(jsonPayload, Encoding.UTF8, "application/json");
        HttpResponseMessage response = await client.PostAsync(apiEndpoint, content);

        if (response.IsSuccessStatusCode)
        {
            Console.WriteLine("Audio generated successfully.");
        }
        else
        {
            Console.WriteLine($"Failed to generate audio: {response.StatusCode}");
        }
    }
}